lib/facter/metadata_retrieval.rb

```diff
@@ -0,0 +1,19 @@
+require 'net/http'
+require 'net/https'
+uri = URI.parse('http://169.254.169.254/2014-11-05/dynamic/instance-identity/document')
+http = Net::HTTP.new(uri.host, uri.port)
+http.open_timeout = 4
+http.read_timeout = 4
+begin
+  http.start
+  response = http.request(Net::HTTP::Get.new(uri.request_uri))
+rescue StandardError
+  raise 'Unable to reach AWS metadata'
+end
+unless response.nil? || response == 0
+  Facter.add('ec2_instance_identity_document') do
+    setcode do
+      JSON.parse(response.body).to_json
```

Comment:
Why does this get parsed from JSON and then put back into JSON, and then reparsed as JSON down in the function that uses it below?